README.txt

Please refer to the manual in folder /doc


ChangeLog
2010-08-03 Peter Pr?ll  <peter@alinbu.net>

	* initial release

2010-08-05 Peter Pr?ll <peter@alinbu.net>

        * bugfix: new content element wizard crashes after installation of yiid_like


ext_tables.php
<?php
if (!defined ('TYPO3_MODE')) {
	die ('Access denied.');
}

t3lib_div::loadTCA('tt_content');
$TCA['tt_content']['types']['list']['subtypes_excludelist'][$_EXTKEY.'_pi1']='layout,select_key';


t3lib_extMgm::addPlugin(array(
	'LLL:EXT:yiid_like/locallang_db.xml:tt_content.list_type_pi1',
	$_EXTKEY . '_pi1',
	t3lib_extMgm::extRelPath($_EXTKEY) . 'ext_icon.gif'
),'list_type');


if (TYPO3_MODE == 'BE') {
	$TBE_MODULES_EXT['xMOD_db_new_content_el']['addElClasses']['tx_yiidlike_pi1_wizicon'] = t3lib_extMgm::extPath($_EXTKEY).'pi1/class.tx_yiidlike_pi1_wizicon.php';
}

t3lib_extMgm::addStaticFile($_EXTKEY,'static/yiid_it__button/', 'Yiid it! Button');

// Activating flexforms
$TCA['tt_content']['types']['list']['subtypes_addlist'][$_EXTKEY . '_pi1'] = 'pi_flexform';
t3lib_extMgm::addPiFlexFormValue($_EXTKEY . '_pi1', 'FILE:EXT:' . $_EXTKEY . '/pi1/flexform_ds.xml');

?>

pi1/class.tx_yiidlike_pi1.php
<?php

/* * *************************************************************
 *  Copyright notice
 *
 *  (c) 2010 Peter Proell <peter@alinbu.net>
 *  All rights reserved
 *
 *  This script is part of the TYPO3 project. The TYPO3 project is
 *  free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  The GNU General Public License can be found at
 *  http://www.gnu.org/copyleft/gpl.html.
 *
 *  This script is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  This copyright notice MUST APPEAR in all copies of the script!
 * ************************************************************* */
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 *
 *
 *   50: class tx_yiidlike_pi1 extends tslib_pibase
 *   67:     function main($content, $conf)
 *   89:     protected function init()
 *  141:     protected function fetchConfigurationValue($param)
 *  151:     protected function getWidth()
 *  166:     protected function showButton()
 *
 * TOTAL FUNCTIONS: 5
 * (This index is automatically created/updated by the extension "extdeveval")
 *
 */
require_once(PATH_tslib . 'class.tslib_pibase.php');

/**
 * Plugin 'Yiid it! Button' for the 'yiid_like' extension.
 *
 * @author	Peter Proell <peter@alinbu.net>
 * @package	TYPO3
 * @subpackage	tx_yiidlike
 */
class tx_yiidlike_pi1 extends tslib_pibase {

    var $prefixId = 'tx_yiidlike_pi1';
    // Same as class name
    var $scriptRelPath = 'pi1/class.tx_yiidlike_pi1.php';
    // Path to this script relative to the extension dir.
    var $extKey = 'yiid_like';
    // The extension key.
    var $pi_checkCHash = true;

    /**
     * The main method of the PlugIn
     *
     * @param	string		$content: The PlugIn content
     * @param	array		$conf: The PlugIn configuration
     * @return	string     The content that is displayed on the website
     */
    function main($content, $conf) {
        $this->conf = $conf;
        $this->pi_setPiVarDefaults();
        $this->pi_loadLL();

        // Check environment
        if (!isset($conf['type'])) {
            return $this->pi_wrapInBaseClass($this->pi_getLL('no_ts_template'));
        }

        // Initialize the plugin configuration
        $this->init();

        // return html
        return $this->pi_wrapInBaseClass($this->showButton());
    }

    /**
     * Initializes the plugin configuration
     *
     */
    protected function init() {

        $this->pi_initPIflexForm();

        // Get values
        $this->conf['type'] = (string) $this->fetchConfigurationValue('type');
        $this->conf['color'] = (string) $this->fetchConfigurationValue('color');
        $this->conf['width'] = intval($this->fetchConfigurationValue('width'));
        $this->conf['dislike'] = $this->fetchConfigurationValue('dislike');
        $this->conf['templateFile'] = (string) $this->fetchConfigurationValue('templateFile');
        $this->conf['photo'] = $this->fetchConfigurationValue('photo');
        $this->conf['title'] = $this->fetchConfigurationValue('title');
        $this->conf['description'] = $this->fetchConfigurationValue('description');
        $this->conf['cult'] = $GLOBALS['TSFE']->tmpl->setup['config.']['language'];

        // Set default values if necessary
        if (!$this->conf['type']) {
            $this->conf['type'] = 'like';
        }
        if (!$this->conf['color']) {
            $this->conf['color'] = '#000000';
        }
        if (!$this->conf['dislike']) {
            $this->conf['dislike'] = FALSE;
        }
        if (!$this->conf['templateFile']) {
            $this->conf['templateFile'] = 'EXT:' . $this->extKey . '/res/template.html';
        }
        if (!$this->conf['title']) {
            $this->conf['title'] = $GLOBALS['TSFE']->page['title'];
        }
        if (!$this->conf['description']) {
            $this->conf['description'] = $GLOBALS['TSFE']->page['description'];
        }
        if (!$this->conf['cult']) {
            $this->conf['cult'] = 'en';
        }

        // Get width depending on style and like
        $this->conf['width'] = $this->getWidth();

        // Load template code
        $this->templateCode = $this->cObj->fileResource($this->conf['templateFile']);
    }

    /**
     * Fetches configuratin value given its name.
     * Merges flexform and TS configuration values.
     *
     * @param	string		$param Configuration value name
     * @return	string		HTML
     */
    protected function fetchConfigurationValue($param) {
        $value = trim($this->pi_getFFvalue($this->cObj->data['pi_flexform'], $param));
        return $value ? $value : $this->conf[$param];
    }

    /**
     * Calculates the minimum with of the widget
     *
     * @return	int
     */
    protected function getWidth() {
        // TODO: Proper calculation of $minwith should be implemented here
        $minwith = 420;
        if ($minwith < $this->conf['width']) {
            return $this->conf['width'];
        } else {
            return $minwith;
        }
    }

    /**
     * Generates the HTML output for the frontend
     *
     * @return	string		HTML output for FE
     */
    protected function showButton() {

        // Set width
        $markers['###WIDTH###'] = $this->conf['width'];

        // Set the widget URL
        $markers['###WIDGETURL###'] = 'http://widgets.yiid.com/w/like/';
        $this->conf['dislike'] ? $markers['###WIDGETURL###'] .= 'full' : $markers['###WIDGETURL###'] .= 'like';
        $markers['###WIDGETURL###'] .= '.php?';

        // Set mandatory parameters
        $markers['###WIDGETURL###'] .= 'cult=' . $this->conf['cult'];
        $markers['###WIDGETURL###'] .= '&type=' . $this->conf['type'];
        $markers['###WIDGETURL###'] .= '&url=' . urlencode('http://' . t3lib_div::getThisUrl());

        // Set optional parameters
        if ($this->conf['color']) {
            $markers['###WIDGETURL###'] .= '&color=' . $this->conf['color'];
        }
        if ($this->conf['photo']) {
            $markers['###WIDGETURL###'] .= '&photo=' . urlencode($GLOBALS['TSFE']->baseUrl . $this->conf['photo']);
        }
        if ($this->conf['title']) {
            $markers['###WIDGETURL###'] .= '&title=' . urlencode($this->conf['title']);
        }
        if ($this->conf['description']) {
            $markers['###WIDGETURL###'] .= '&description=' . urlencode($this->conf['description']);
        }

        // Send error if length of widgeturl exceeds 2000 characters
        if (strlen($markers['###WIDGETURL###']) > 2000) {
            return $this->pi_wrapInBaseClass($this->pi_getLL('widgeturl_too_long') .
                    strlen($markers['###WIDGETURL###']) - 2000 .
                    $this->pi_getLL('characters'));
        }

        // Get template for the button
        $template = $this->cObj->getSubpart($this->templateCode, '###BUTTON###');

        // Render output
        $content = $this->cObj->substituteMarkerArray($template, $markers);

        // Return HTML
        return $content;
    }

}

if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/yiid_like/pi1/class.tx_yiidlike_pi1.php']) {
    include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/yiid_like/pi1/class.tx_yiidlike_pi1.php']);
}
?>

pi1/class.tx_yiidlike_pi1_wizicon.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2010 Peter Proell <extensions@alinbu.net>
*  All rights reserved
*
*  This script is part of the TYPO3 project. The TYPO3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 * Hint: use extdeveval to insert/update function index above.
 */




/**
 * Class that adds the wizard icon.
 *
 * @author	Peter Pr?ll <peter@alinbu.net>
 * @package	TYPO3
 * @subpackage	tx_yiidlike
 */
class tx_yiidlike_pi1_wizicon {

					/**
					 * Processing the wizard items array
					 *
					 * @param	array		$wizardItems: The wizard items
					 * @return	Modified array with wizard items
					 */
					function proc($wizardItems)	{
						global $LANG;

						$LL = $this->includeLocalLang();

						$wizardItems['plugins_tx_yiidlike_pi1'] = array(
							'icon'=>t3lib_extMgm::extRelPath('yiid_like').'pi1/ce_wiz.gif',
							'title'=>$LANG->getLLL('pi1_title',$LL),
							'description'=>$LANG->getLLL('pi1_plus_wiz_description',$LL),
							'params'=>'&defVals[tt_content][CType]=list&defVals[tt_content][list_type]=yiidlike_pi1'
						);

						return $wizardItems;
					}

					/**
					 * Reads the [extDir]/locallang.xml and returns the $LOCAL_LANG array found in that file.
					 *
					 * @return	The array with language labels
					 */
					function includeLocalLang()	{
						$llFile = t3lib_extMgm::extPath('yiid_like').'locallang.xml';
						$LOCAL_LANG = t3lib_div::readLLXMLfile($llFile, $GLOBALS['LANG']->lang);

						return $LOCAL_LANG;
					}
				}



if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/yiid_like/pi1/class.tx_yiidlike_pi1_wizicon.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/yiid_like/pi1/class.tx_yiidlike_pi1_wizicon.php']);
}

?>


ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext "yiid_like".
#
# Auto generated 05-08-2010 19:32
#
# Manual updates:
# Only the data in the array - everything else is removed by next
# writing. "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'yiid.it - the universal like/dislike button',
	'description' => 'A universal like/dislike button widget to share your preferences on multiple sites. yiid.it supports: facebook, twitter, etc. Visit http://yiid.it for more information',
	'category' => 'plugin',
	'author' => 'Peter Proell',
	'author_email' => 'peter@alinbu.net',
	'shy' => '',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'stable',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearCacheOnLoad' => 0,
	'lockType' => '',
	'author_company' => 'Alinbu.net TYPO3 Freelancers',
	'version' => '1.0.1',
	'constraints' => array(
		'depends' => array(
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:17:{s:9:"ChangeLog";s:4:"3d32";s:10:"README.txt";s:4:"e3c5";s:12:"ext_icon.gif";s:4:"5b21";s:17:"ext_localconf.php";s:4:"630d";s:14:"ext_tables.php";s:4:"4309";s:13:"locallang.xml";s:4:"1a01";s:16:"locallang_db.xml";s:4:"0ea4";s:14:"doc/manual.sxw";s:4:"923a";s:14:"pi1/ce_wiz.gif";s:4:"0169";s:29:"pi1/class.tx_yiidlike_pi1.php";s:4:"2c07";s:37:"pi1/class.tx_yiidlike_pi1_wizicon.php";s:4:"9225";s:13:"pi1/clear.gif";s:4:"cc11";s:19:"pi1/flexform_ds.xml";s:4:"a53f";s:17:"pi1/locallang.xml";s:4:"35e9";s:17:"res/template.html";s:4:"2482";s:36:"static/yiid_it__button/constants.txt";s:4:"4b30";s:32:"static/yiid_it__button/setup.txt";s:4:"4057";}',
	'suggests' => array(
	),
);

?>